src/runtime/components/elements/Alert.vue

```diff
@@ -1,8 +1,12 @@
 <template>
   <div :class="alertClass" v-bind="attrs">
     <div class="flex" :class="[ui.gap, { 'items-start': (description || $slots.description), 'items-center': !description && !$slots.description }]">
-      <UIcon v-if="icon" :name="icon" :class="ui.icon.base" />
-      <UAvatar v-if="avatar" v-bind="{ size: ui.avatar.size, ...avatar }" :class="ui.avatar.base" />
+      <slot name="icon" :icon="icon" :base="ui.icon.base">
```

Comment:
Really interesting idea to pass down `ui.icon.base` to the slot. We should do this everywhere! Not sure `base` is the best name though. Could we pass `class`?

Reply:
Just a suggestion but maybe pass an `ui` object:

- `<slot name="icon" :icon="icon" :ui="ui.icon">`
- `<slot name="avatar" :avatar="avatar" :ui="ui.avatar">`